pyproject.toml

```
@@ -42,6 +42,71 @@ classifiers = [
     "Programming Language :: Python :: 3 :: Only",
     "Topic :: Scientific/Engineering :: Physics",
 ]
+[dependency-groups]
+ci = [
+    "numpy",
+    "scipy",
+    "cvxpy",
+    "cvxopt",
+    "networkx",
+    "rustworkx",
+    "autograd",
+    "tomlkit",
+    "appdirs",
+    "packaging",
+    "autoray>=0.8.0",
+    "matplotlib",
+    "requests",
+    "rich",
+    "diastatic-malt",
+    "pandas",
+    "build"
```

I believe everything else is already included as core dependencies of PL.
```suggestion
    "cvxpy",
    "cvxopt",
    "matplotlib",
    "rich",
    "pandas",
    "build"
```